nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/main/java/org/apache/nifi/lookup/RestLookupService.java

```diff
@@ -0,0 +1,390 @@
+        RequestBody requestBody = null;
+        if (body != null) {
+            requestBody = RequestBody.create(mt, body);
+        }
+        Request.Builder request = new Request.Builder()
+                .url(endpoint);
+        switch(method.toLowerCase()) {
+            case "delete":
+                request = body != null ? request.delete(requestBody) : request.delete();
+                break;
+            case "get":
+                request = request.get();
+                break;
+            case "post":
+                request = request.post(requestBody);
+                break;
+            case "put":
+                request = request.put(requestBody);
+                break;
+        }
+
+        if (headers != null) {
+            for (Map.Entry<String, String> header : headers.entrySet()) {
+                request = request.addHeader(header.getKey(), header.getValue());
+            }
+        }
+
+        if (!basicUser.isEmpty() && !isDigest) {
+            String credential = Credentials.basic(basicUser, basicPass);
+            request = request.header("Authorization", credential);
+        }
+
+        return request.build();
+    }
+
+    private String basicUser;
+    private String basicPass;
+    private boolean isDigest;
+
+    private void setAuthenticator(OkHttpClient.Builder okHttpClientBuilder, ConfigurationContext context) {
+        final String authUser = trimToEmpty(context.getProperty(PROP_BASIC_AUTH_USERNAME).getValue());
+        this.basicUser = authUser;
+
+
+        isDigest = context.getProperty(PROP_DIGEST_AUTH).asBoolean();
+        // If the username/password properties are set then check if digest auth is being used
+        if (!authUser.isEmpty() && isDigest) {
+            final String authPass = trimToEmpty(context.getProperty(PROP_BASIC_AUTH_PASSWORD).getValue());
+            this.basicPass = authPass;
+
+            /*
+             * OkHttp doesn't have built-in Digest Auth Support. A ticket for adding it is here[1] but they authors decided instead to rely on a 3rd party lib.
+             *
+             * [1] https://github.com/square/okhttp/issues/205#issuecomment-154047052
+             */
+            final Map<String, CachingAuthenticator> authCache = new ConcurrentHashMap<>();
+            com.burgstaller.okhttp.digest.Credentials credentials = new com.burgstaller.okhttp.digest.Credentials(authUser, authPass);
+            final DigestAuthenticator digestAuthenticator = new DigestAuthenticator(credentials);
+
+            okHttpClientBuilder.interceptors().add(new AuthenticationCacheInterceptor(authCache));
+            okHttpClientBuilder.authenticator(new CachingAuthenticatorDecorator(digestAuthenticator, authCache));
+        }
+    }
+
+    @Override
+    public Class<?> getValueType() {
+        return Record.class;
+    }
+
+    @Override
+    public Set<String> getRequiredKeys() {
+        return new HashSet<String>(){{
```

Comment:
We should avoid the creation of anonymous inner classes when we can. We can also make this an UnmodifiableSet and just create a single instance of it as a member variable instead of constantly creating the Set

Reply:
Done.